### PR TITLE
bug 1781804: fix revision pending message

### PR DESCRIPTION
**Pull request by deads2k**
I needed a correct message to figure out what state the operator thinks it is in

/assign @sttts

### CODE REVIEW

**Comment by deads2k** [comment]
/hold

this fixes an immediate problem, but the entire approach looks wrong.

File: pkg/operator/staticpod/controller/installer/installer_controller.go [hunk 1]
```diff
@@ -537,11 +536,11 @@ func (c *InstallerController) newNodeStateForInstallInProgress(currNodeState *op
 
 	switch installerPod.Status.Phase {
 	case corev1.PodSucceeded:
-		if newRevisionPending {
+		if pendingNewRevision := latestRevisionAvailable > currNodeState.TargetRevision; pendingNewRevision {
 			// stop early, don't wait for ready static pod because a new revision is waiting
 			ret.LastFailedRevision = currNodeState.TargetRevision
 			ret.TargetRevision = 0
-			ret.LastFailedRevisionErrors = []string{fmt.Sprintf("static pod of revision has been installed, but is not ready while new revision %d is pending", currNodeState.TargetRevision)}
+			ret.LastFailedRevisionErrors = []string{fmt.Sprintf("static pod of revision %d has been installed, but is not ready while new revision %d is pending", currNodeState.TargetRevision, latestRevisionAvailable)}
```
Comment (sttts) [inline pkg/operator/staticpod/controller/installer/installer_controller.go]: oops.

**Comment by deads2k** [comment]
/hold cancel

the approach is weird (we aren't degraded for this reason), but we are running

File: pkg/operator/staticpod/controller/installer/installer_controller.go (hunk 1 above)
Comment (mfojtik) [reply pkg/operator/staticpod/controller/installer/installer_controller.go]: i thought go vet catch these (I fixed some in past)

**Pull request merged**